### PR TITLE
Update compatibility notes

### PR DESCRIPTION
**Pull request by infoverload**
## Summary of the changes / Why this improves CrateDB

This page is out of date.

https://github.com/crate/tech-writing-domain/issues/248

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/compatibility.rst [hunk 1]
```diff
@@ -89,19 +88,17 @@ the record is modified. This version number can be used to implement patterns
...
+These **features** of `standard SQL`_ are not supported:
 
-- Stored Procedures
+- Stored procedures
 
 - Triggers
 
-- ``VALUES`` list used as constant tables
-
-- ``WITH`` Statements
```
Comment (mfussenegger) [inline docs/appendices/compatibility.rst:103]: I think this was referring to [common table expressions](https://www.postgresql.org/docs/current/queries-with.html) which we still do not support.

File: docs/appendices/compatibility.rst [hunk 2]
```diff
@@ -89,19 +88,17 @@ the record is modified. This version number can be used to implement patterns
 like :ref:`sql_occ`, which can be used to solve many of the use cases that
 would otherwise require traditional transactions.
 
+Transaction statements do not throw an error.
```
Comment (mfussenegger) [inline docs/appendices/compatibility.rst:91]: ``ROLLBACK`` will result in an error, the others are basically no-ops

**Review by mechanomi — COMMENTED** [review]
changes look good (with @mfussenegger's comments in mind)

though as I said on the issue, I think this content would be better merged into the PostgreSQL wire protocol document

**Review by mechanomi — APPROVED** [review]
I thought I found a few minor corrections but each time I was wrong haha. perfecto! 👌

**Pull request merged**